File: Education/src/main/java/com/Test/Education/services/EtudiantService.java
package com.Test.Education.services;

import com.Test.Education.dto.MoyDto;
import com.Test.Education.models.Enseignant;
import com.Test.Education.models.Etudiant;
import com.Test.Education.models.User;

import java.util.List;

public interface EtudiantService {
    Etudiant save(Etudiant e);
    void deleteEtudiant(Long id );
    List<Etudiant> findAll();
    Etudiant findById(Long id);
    Etudiant updateEtudiant (Etudiant e, Long id);
   Etudiant findByUser (User user);
   MoyDto calculMoy (Etudiant e );
    
    
}

File: Front/Education/src/app/services/etudiant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Etudiant } from '../models/Etudiant';
import { User } from '../models/User';
import { MoyDto } from '../models/MoyDto';

@Injectable({
  providedIn: 'root'
})
export class EtudiantService {

  constructor(private http:HttpClient) { }

  addEtudiant(e:Etudiant){
    return this.http.post<Etudiant>("http://localhost:8080/auth/signupEtudiant",e);
  }
  findAll(){
    return this.http.get<Etudiant[]>("http://localhost:8080/etudiant/all");
  }
  deleteE(id:number){
    return this.http.delete<Etudiant>("http://localhost:8080/etudiant/delete/"+id);
  }
  findById(id:number){
    return this.http.get<Etudiant>("http://localhost:8080/etudiant/find/"+id);
  }
  Edit(etud:Etudiant,id:number){
    return this.http.put<Etudiant>("http://localhost:8080/etudiant/update/"+id,etud);
  }
  finfByUser(user:User){
    return this.http.post<Etudiant>("http://localhost:8080/etudiant/find/",user);
  }
  CalculMoy(etud:Etudiant){
    return this.http.post<MoyDto>("http://localhost:8080/etudiant/moyenne/",etud);
  }

}
